src/processors/universal.ts

```diff
@@ -370,6 +360,108 @@ class UniversalProcessor extends BaseOrderProcessor {
     return unlockTx;
   }
 
+  private async unlockOrder(
+    orderId: string,
+    order: OrderData,
+    context: OrderProcessorContext
+  ) {
+    if (this.isUnlockBatchLocked) {
+      setTimeout(() => {
+        this.unlockOrder(orderId, order, context);
+      }, 0);
+      return;
+    }
+    const logger = context.logger;
+    this.isUnlockBatchLocked = true;
+    try {
+      logger.info(`Unlocking is started`);
+      const key = `${order.give.chainId}_${order.take.chainId}`;
+      let orderIds = this.unlockBatchesOrderIdMap.get(key);
+      if (!orderIds) {
+        orderIds = new Set();
+      }
+      orderIds.add(orderId);
+      this.unlockBatchesOrderIdMap.set(key, orderIds);
+
+      if (orderIds.size !== this.UNLOCK_BATCH_SIZE) {
+        logger.debug(`Batch for unlock is not full`);
```

Comment:
This check is not needed. We just need to start the batch-unlock process only when there is enough orders in the map.

Reply:
Move the batch-unlock code to another private method.

The better pseudo-code for `unlockOrder`:

```
func unlockOrder(order) {
  if (isBatchUnlockLocked) {
    unlockBatches[order.give.chain].append(order)
    return
  }

  if (unlockBatches[order.give.chain].size >= BATCH_SIZE) {
    this.processBatches(unlockBatches[order.give.chain])
  }
}